mainfile

```diff
@@ -196,10 +216,40 @@ void NormalRouteForCarOnCircut(void){
 
 }
 
-// THIS METHOD MAKES THE CAR RUN IN A CIRCLE
-void GoInCircle(diameter){
-  diameter = Math.random(<10);
-  circumference = double((PI)diameter);
+// THIS METHOD CONVERTS ULTRASONIC SENSOR INPUT TO INCHES
+long microsecondsToInches(long microseconds){
+    // According to Parallax's datasheet for the PING))), there are 73.746
+  // microseconds per inch (i.e. sound travels at 1130 feet per second).
+  // This gives the distance travelled by the ping, outbound and return,
+  // so we divide by 2 to get the distance of the obstacle.
+  // See: https://www.parallax.com/package/ping-ultrasonic-distance-sensor-downloads/
+  return microseconds / 74 / 2;
+}
+
+// THIS METHOD CONVERTS ULTRASONIC SENSOR INPUT TO CENTIMETERS
+long microsecondsToCentimeters(long microseconds){
+  // The speed of sound is 340 m/s or 29 microseconds per centimeter.
+  // The ping travels out and back, so to find the distance of the object we
+  // take half of the distance travelled.
+  return microseconds / 29 / 2;
+}
+
+//Units in/sec
+int speedOfCar(int CurSpeedSetting){
+// Wheel Diameter is 1.5 in
+// Highest RPM on Dc Moter is 255
+// Length of one rotation convert rpm to feet per second
+// Angular velocity is equal to 6 x rpm which gets units degrees per second
+}
+
+long distanceTraveledByCar{
+// Using some kinematic equations 𝑑=𝑣𝑖𝑡+12𝑎𝑡2
+
+
+}
+
+long timeTraveledByCarPriorToBreak{
+
```

Comment:
// i think we might need a method to determine how long it takes to turn a certain number of degrees so turnTime is the method with a parameter of degrees and it will calculate how many seconds per degree or something like that. we can also just test this by having it turn 360 degrees, time it, divide by 360 and store that value in 'double turn' //
double turn = xxxx;
double turnTime(int degrees){
     return turn * degrees;
}
void moveForward(double distance){
for (int i = 0; i < calculateDistance(distance); i++){
     analogWrite(motor1,2,3,4, normal);
     ...
}
}
void turnRight(int degrees){
for (int i = 0; i < turnTime(degrees); i++){
analogWrite(motor 1,3, normal);
analogWrite(motor2,4, slow);
}
}
void turnLeft(int degrees){
for (int i = 0; i < turnTime(degrees); i++){
analogWrite(motor 1,3, slow);
analogWrite(motor2,4, normal);
}
}
// do we want to manually call each method for each segment of the route? i cant really think of another way to do it 
moveForward(time);
moveRight(degrees);
moveLeft(degrees);
etc.... according to the sketch

Reply:
> // i think we might need a method to determine how long it takes to turn a certain number of degrees so turnTime is the method with a parameter of degrees and it will calculate how many seconds per degree or something like that. we can also just test this by having it turn 360 degrees, time it, divide by 360 and store that value in 'double turn' //
> double turn = xxxx;
> double turnTime(int degrees){
> return turn * degrees;
> }
> void moveForward(double distance){
> for (int i = 0; i < calculateDistance(distance); i++){
> analogWrite(motor1,2,3,4, normal);
> ...
> }
> }
> void turnRight(int degrees){
> for (int i = 0; i < turnTime(degrees); i++){
> analogWrite(motor 1,3, normal);
> analogWrite(motor2,4, slow);
> }
> }
> void turnLeft(int degrees){
> for (int i = 0; i < turnTime(degrees); i++){
> analogWrite(motor 1,3, slow);
> analogWrite(motor2,4, normal);
> }
> }
> // do we want to manually call each method for each segment of the route? i cant really think of another way to do it
> moveForward(time);
> moveRight(degrees);
> moveLeft(degrees);
> etc.... according to the sketch